docs/via_guides/data-flow.md

```diff
@@ -0,0 +1,872 @@
+  publication process
+- The dispatcher extracts the proof data and prepares it for publication to the data availability layer
+
+This monitoring approach creates a seamless connection between the prover subsystem and the sequencer's commitment
+process, despite their indirect communication model.
+
+#### Proof Publication Pipeline
+
+Once a proof is detected, the dispatcher follows these steps:
+
+1. Validates the proof data for completeness and format correctness
+2. Publishes the proof to the data availability layer
+3. Creates database entries that serve as indirect tasks for the `via_btc_sender` component
+4. Monitors the publication status to ensure successful commitment
+
+As noted in the codebase, these "tasks" are implemented through database table entries rather than direct communication.
+The dispatcher writes to specific tables, which are then queried by other components looking for work to perform. This
+pattern follows the indirect communication principle established in the system architecture.
+
+**Database Query Example**:
+
+The DA dispatcher uses specialized queries to identify proofs ready for dispatch:
+
+```sql
+SELECT
+    proof_generation_details.l1_batch_number,
+    proof_generation_details.proof_blob_url
+FROM
+    proof_generation_details
+WHERE
+    proof_generation_details.status = 'generated'
+    AND proof_generation_details.proof_blob_url IS NOT NULL
+    AND EXISTS (
+        SELECT 1 FROM via_data_availability
+        WHERE l1_batch_number = proof_generation_details.l1_batch_number
+        AND is_proof = FALSE AND blob_id IS NOT NULL
+    )
+    AND NOT EXISTS (
+        SELECT 1 FROM via_data_availability
+        WHERE l1_batch_number = proof_generation_details.l1_batch_number
+        AND is_proof = TRUE AND blob_id IS NOT NULL
+    )
+ORDER BY proof_generation_details.l1_batch_number
+LIMIT $1
+```
+
+This query demonstrates how complex decision-making is encoded in database queries rather than direct component
+communication.
+
+#### Bitcoin Inscription of Proofs
+
+The actual inscription of proof data on Bitcoin is handled by the [`via_btc_sender`](../../core/node/via_btc_sender)
+component, which:
+
+- Identifies pending proof commitment tasks by querying relevant database tables
+- Constructs Bitcoin inscription messages containing the proof commitment data
+- Creates and signs Bitcoin transactions to inscribe the proof data
+- Broadcasts these transactions to the Bitcoin network
+- Updates the database with transaction confirmation information
+
+The proof inscription process follows the same two-transaction approach (commit and reveal) as used for batch
+commitments, with specialized formatting for proof data.
+
+#### Proof Commitment Data Structure
+
+The proof commitment inscription contains several critical pieces of information:
+
+- Cryptographic proof data verifying the correctness of state transitions
```

Comment:
In the current solution we are not actually submitting the proof data to the Bitcoin to reduce fees, right?
So, just a question if we want to describe the current state or something that we are aiming for?